core/src/main/java/com/datastax/oss/driver/api/core/config/ProgrammaticDriverConfigLoader.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.config;
+
+import com.datastax.oss.driver.internal.core.config.typesafe.DefaultDriverConfigLoader;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigValueFactory;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class ProgrammaticDriverConfigLoader {
+
+  private ProgrammaticDriverConfigLoader() {}
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+
+    private Config config = ConfigFactory.empty();
+
+    private Supplier<Config> fallbackSupplier = DefaultDriverConfigLoader.DEFAULT_CONFIG_SUPPLIER;
+
+    private Builder() {}
+
+    public Builder withBoolean(DriverOption option, boolean value) {
+      return with(option, value);
+    }
+
+    public Builder withBoolean(String profileName, DriverOption option, boolean value) {
+      return with(profileName, option, value);
+    }
+
+    public Builder withBooleanList(DriverOption option, List<Boolean> value) {
+      return with(option, value);
```

Comment:
This is the same as in `TypesafeDriverConfigProfile`. You could extract an internal interface with default methods and only have to implement `with()`. As a bonus, this would help validate that we're not missing any method here.
(provided that the loader becomes an internal class as suggested in my other comment)